cpp/src/io/parquet/page_enc.cu

```diff
@@ -1745,9 +1824,9 @@ dremel_data get_dremel_data(column_view h_col,
   {
     // At this point, curr_col contains the leaf column. Max nesting level is
     // nesting_levels.size().
-    size_t level              = nesting_levels.size() - 1;
+    size_t level              = nesting_levels.size() - 2;
```

Comment:
A change that was made in this PR is to include the leaf level also in `nesting_levels`. All else remains the same. So with the one optimization for single level lists, this becomes a process that goes like this: merge the leaf level def/rep with the immediate parent. The leaf level is the element at `nesting_levels[nesting_levels.size() - 1]` which is merged with the list level at `nesting_levels[nesting_levels.size() - 2]` and so on.

Reply:
I think it would be useful to improve the comment here.  Same for the -3 below.  A description of why these numbers the way they are will definitely ease comprehension for someone coming in in the future.